packages/ra-ui-materialui/src/field/ArrayField.tsx

```diff
@@ -153,15 +153,14 @@ export const ArrayField: FunctionComponent<
     });
 };
 
-const EnhancedArrayField = pure<ArrayFieldProps & FieldProps>(ArrayField);
+const EnhancedArrayField: FunctionComponent<ArrayFieldProps & FieldProps> = memo(ArrayField);
```

Comment:
We don't even need the `EnhancedArrayField`: place the `memo` at the `ArrayField` definition.

Reply:
Done. I hope.